plugins/mongodb-adapter/src/main/java/org/polypheny/db/adapter/mongodb/rules/MongoProject.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2019-2023 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.adapter.mongodb.rules;
+
+
+import com.google.common.collect.Streams;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import lombok.Getter;
+import org.bson.BsonDocument;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+import org.polypheny.db.adapter.mongodb.MongoAlg;
+import org.polypheny.db.adapter.mongodb.bson.BsonFunctionHelper;
+import org.polypheny.db.adapter.mongodb.util.RexToMongoTranslator;
+import org.polypheny.db.algebra.AlgNode;
+import org.polypheny.db.algebra.constant.Kind;
+import org.polypheny.db.algebra.core.Project;
+import org.polypheny.db.algebra.metadata.AlgMetadataQuery;
+import org.polypheny.db.algebra.type.AlgDataType;
+import org.polypheny.db.algebra.type.AlgDataTypeFactory;
+import org.polypheny.db.algebra.type.AlgDataTypeField;
+import org.polypheny.db.catalog.logistic.DataModel;
+import org.polypheny.db.plan.AlgCluster;
+import org.polypheny.db.plan.AlgOptCost;
+import org.polypheny.db.plan.AlgPlanner;
+import org.polypheny.db.plan.AlgTraitSet;
+import org.polypheny.db.rex.RexCall;
+import org.polypheny.db.rex.RexIndexRef;
+import org.polypheny.db.rex.RexNode;
+import org.polypheny.db.rex.RexVisitorImpl;
+import org.polypheny.db.util.Pair;
+import org.polypheny.db.util.Util;
+
+
+/**
+ * Implementation of {@link Project} relational expression in MongoDB.
+ */
+public class MongoProject extends Project implements MongoAlg {
+
+    public MongoProject( AlgCluster cluster, AlgTraitSet traitSet, AlgNode input, List<? extends RexNode> projects, AlgDataType rowType ) {
+        super( cluster, traitSet, input, projects, adjustRowType( rowType, projects, input ) );
+        assert getConvention() == CONVENTION;
+        //assert getConvention() == input.getConvention(); // TODO DL fix logicalFilter bug
```

Comment:
Is this fixed?
It might make sense to activate asserts for (some) of the test runs.

Reply:
This is an artifact as no other physical implementations use this. It does not really make sense, as it blocks the creation of more complex algebra trees during optimizations.